src/main/java/com/aws/greengrass/crypto/CryptoI.java

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.aws.greengrass.crypto;
+
+import com.aws.greengrass.config.Subscriber;
+import com.aws.greengrass.config.WhatHappened;
+import com.aws.greengrass.deployment.DeviceConfiguration;
+import com.aws.greengrass.util.Coerce;
+import com.aws.greengrass.util.EncryptionUtils;
+import com.aws.greengrass.util.exceptions.TLSAuthException;
+import software.amazon.awssdk.iot.AwsIotMqttConnectionBuilder;
+
+import java.io.IOException;
+import java.security.GeneralSecurityException;
+import java.security.KeyStore;
+import java.security.PrivateKey;
+import java.security.cert.Certificate;
+import java.security.cert.X509Certificate;
+import java.util.List;
+import javax.inject.Inject;
+import javax.net.ssl.KeyManagerFactory;
+import javax.net.ssl.TrustManagerFactory;
+import javax.net.ssl.X509KeyManager;
+import javax.net.ssl.X509TrustManager;
+import javax.security.auth.x500.X500Principal;
+
+public interface CryptoI {
+    AwsIotMqttConnectionBuilder getIotMqttConnectionBuilder();
+
+    X509KeyManager[] getKeyManagers() throws TLSAuthException;
+
+    X509TrustManager[] getTrustManagers() throws TLSAuthException;
+
+    void setProviderChanged(Runnable providerChanged);
```

Comment:
If we expect implementations to handle when a provider changes, can't this action be correlated? For example: `cleanup` which is invoked by the provider when the crypto implementation is updated?

Reply:
The implementations are only expected to notify the CryptoProvider that _they_ (the implementation) has changed and then the CryptoProvider will notify the users.

Reply:
Probably needs to be renamed since I see why you're confused.